### PR TITLE
Upgrade Gradle and nebula.ospackage_version

### PR DESCRIPTION
**Pull request by cliu123**
##  opendistro-for-elasticsearch/security pull request intake form
_Please provide as much details as possible to get feedback/acceptance on your PR quickly_

 
 1. __Category:__ (_Enhancement, New feature, Bug fix, Test fix, Refactoring, Maintenance, Documentation_)
 Maintenance
 
 
 2. __Github Issue # or road-map entry, if available:__
Upgrade Gradle and nebula.ospackage version

 3. __Description of changes:__
It's necessary for build system migration from Maven to Gradle

 4. __Why these changes are required?__ 
It's necessary for build system migration from Maven to Gradle

 5. __What is the old behavior before changes and new behavior after changes?__ (_Please add any example/logs/screen-shot if available_)

 6. __Testing done:__ (_Please provide details of testing done: Unit testing, integration testing and manual testing_)
 - UT
- Manual Testing:
     - Run `./gradlew clean build buildDeb buildRpm --no-daemon -ParchivePath=`ls $(pwd)/target/releases/opensearch-security-*.zip | grep -v admin-standalone` -Dbuild.snapshot=false`
     - `deb` and `rpm` packages get generated properly
 
 
 7. __TO-DOs, if any:__ (_Please describe pending items and provide Github issues# for each of them_)

 8. __Is it backport from main branch?__ (_If yes, please add backport PR # and commits #_)

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-4.10.2-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.8-bin.zip
```
Comment (cliu123) [inline gradle/wrapper/gradle-wrapper.properties:3]: @sujithvm `ES 7.10.2/OpenSearch 1.0.0-beta1` doesn't support Gradle 7.0.
https://github.com/elastic/elasticsearch/issues/72604

File: build.gradle [hunk 2]
```diff
@@ -88,10 +88,24 @@ ospackage {
 buildRpm {
     arch = 'NOARCH'
     addParentDirs = false
-    archiveName "${packageName}-${version}.rpm"
+    finalizedBy 'renameRpm'
+    task renameRpm(type: Copy) {
+        from("$buildDir/distributions")
```
Comment (sujithvm) [inline build.gradle:93]: Is it possible to reuse this task for both deb and rpm ?

Comment (cliu123) [reply build.gradle:93]: This is not a custom task class. It doesn't take parameters like methods. I don't see a way to reuse the task.

Comment (sujithvm) [reply build.gradle:93]: You can dynamically create new task as template

Comment (sujithvm) [reply build.gradle:93]: Something like this 

```
buildRpm {
    arch = 'NOARCH'
    addParentDirs = false
    finalizedBy renameTask(archiveName, packageName, 'rpm')
}

buildDeb {
    arch = 'all'
    finalizedBy renameTask(archiveName, packageName, 'deb')
}

def renameTask(archiveName, packageName, arch) {
    return tasks.create("rename${arch}", Copy) {
        from("$buildDir/distributions")
        into("$buildDir/distributions")
        include archiveName
        rename archiveName, "${packageName}-${version}.${arch}"
        doLast { delete file("$buildDir/distributions/$archiveName") }
    }
}
```

Comment (cliu123) [reply build.gradle:90]: @sujithvm I've updated the changes.

**Comment by cliu123** [comment]
Will re-open later if needed.

**Pull request closed without merging**